### PR TITLE
Remove more legacy Runner v1 cruft.

### PR DESCRIPTION
**Pull request by robertwb**
------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/get-started-contributing/#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by robertwb** [comment]
R: @tvalentyn

File: sdks/python/apache_beam/runners/runner.py [hunk 1]
```diff
@@ -158,181 +177,39 @@ def run_pipeline(
...
-    Runners should override this method.
     """
-    raise NotImplementedError
+    pipeline.visit(
+        group_by_key_input_visitor(
+            not options.view_as(TypeOptions).allow_non_deterministic_key_coders)
+    )
+
+    # TODO: https://github.com/apache/beam/issues/19168
+    # portable runner specific default
```
Comment (tvalentyn) [inline sdks/python/apache_beam/runners/runner.py:187]: we can and plan to make this a  default for dataflow as well: https://github.com/apache/beam/issues/26996

File: sdks/python/apache_beam/runners/runner.py [hunk 2]
```diff
@@ -158,181 +177,39 @@ def run_pipeline(
...
+    if options.view_as(SetupOptions).sdk_location == 'default':
+      options.view_as(SetupOptions).sdk_location = 'container'
+
+    return self.run_full_pipeline(
```
Comment (tvalentyn) [inline sdks/python/apache_beam/runners/runner.py:191]: What is the semantic  distinction between `run_pipeline` vs `run_full_pipeline`? It sounds like `run_pipeline` could run exectute subgraphs, but it calls into `run_full_pipeline`, which is supposed to run the entire graph.

**Review by tvalentyn — COMMENTED** [review]
noting that some tests & lint failed on latest snapshot.

File: sdks/python/apache_beam/transforms/environments.py [hunk 3]
```diff
@@ -699,6 +721,26 @@ def default(cls):
     return cls(capabilities=python_sdk_capabilities(), artifacts=())
 
 
+@Environment.register_urn(python_urns.EMBEDDED_PYTHON_LOOPBACK, None)
+class PythonLoopbackEnvironment(EmbeddedPythonEnvironment):
+  """Used as a stub when the loopback worker has not yet been started."""
+  def to_runner_api_parameter(self, context):
```
[the remark this answers is not in the document]
Comment (robertwb) [reply sdks/python/apache_beam/transforms/environments.py:726]: Done.

File: sdks/python/apache_beam/runners/runner.py (hunk 2 above)
Comment (robertwb) [reply sdks/python/apache_beam/runners/runner.py:191]: Mostly it's just a type distinction, but for backwards compatibility and the fact that only names (not type signatures) are used to resolve methods in Python I needed to call it something different. (IIRC, the old version could execute subgraphs at some point, I don't know if anyone uses that capability anymore.)

File: sdks/python/apache_beam/runners/runner.py (hunk 1 above)
Comment (robertwb) [reply sdks/python/apache_beam/runners/runner.py:187]: +1

File: sdks/python/apache_beam/runners/runner.py (hunk 2 above)
Comment (tvalentyn) [reply sdks/python/apache_beam/runners/runner.py:190]: i see, thanks. the only alternative that comes to mind is `run_portable_pipeline()`, but not sure if that would be a better name.

Comment (hjtran) [reply sdks/python/apache_beam/runners/runner.py:190]: Sorry for crashing, but I got tripped up by this already when writing code on top of these changes. I think any of `run_portable_pipeline` / `run_pipeline_proto` / `run_pipeline_from_proto` would be a bit clearer

**Comment by robertwb** [comment]
Ping on this @tvalentyn

File: sdks/python/apache_beam/runners/runner.py (hunk 2 above)
Comment (tvalentyn) [reply sdks/python/apache_beam/runners/runner.py:190]: cc: @robertwb

**Review by tvalentyn — APPROVED** [review]

**Comment by tvalentyn** [comment]
Run Python_Integration PreCommit 3.11

**Pull request merged**